ts/types.go

```diff
@@ -45,27 +50,70 @@ type id struct {
 }
 
 // Data returns the binary representation of an ID
-func (id *id) Data() []byte {
-	return id.data
+func (v *id) Data() []byte {
+	return v.data
 }
 
-func (id *id) Equal(value ID) bool {
-	return bytes.Equal(id.Data(), value.Data())
+func (v *id) Equal(value ID) bool {
+	return bytes.Equal(v.Data(), value.Data())
 }
 
 var null = Hash{}
 
 // Hash returns (and calculates, if needed) the hash for an ID
-func (id *id) Hash() Hash {
-	if bytes.Equal(id.hash[:], null[:]) {
-		id.hash = md5.Sum(id.data)
+func (v *id) Hash() Hash {
+	if bytes.Equal(v.hash[:], null[:]) {
+		v.hash = md5.Sum(v.data)
 	}
 
-	return Hash(id.hash)
+	return Hash(v.hash)
+}
+
+func (v *id) String() string {
+	return string(v.data)
+}
+
+// Clone constructs a new ID identical to the current
+func (v *id) Clone() ID {
+	return &id{data: v.data, hash: v.hash}
+}
+
+// Pool represents an object pool of IDs
+type Pool struct {
+	pool pool.ObjectPool
+}
+
+// NewPool constructs a new Pool
+func NewPool(options pool.ObjectPoolOptions) *Pool {
```

Comment:
`NewIDPool(...)` would be better in case we add a pool for a different type of object in the `ts` package?

Also: we've been returning interface types to ease mocking, for consistency would be good do the same for this pool and unexport the struct.